apps/server/src/modules/task/controller/api-test/task.api.spec.ts

```diff
@@ -1456,4 +1457,256 @@ describe('Task Controller (API)', () => {
+			});
+			const mathsCourse = courseFactory.build({
+				name: 'maths',
+				teachers: [teacher2],
+				students: [student1, student2],
+			});
+			const algebraLesson = lessonFactory.build({
+				name: 'maths',
+				course: mathsCourse,
+			});
+
+			// create tasks
+			const englishTask1 = taskFactory.build({
+				name: 'Write an essay',
+				course: englishCourse,
+				users: [student1, student2],
+				creator: teacher1,
+				finished: [student1],
+			});
+			const englishTask2 = taskFactory.build({
+				name: 'grammer1',
+				creator: teacher1,
+				lesson: grammerLesson,
+				users: [student1],
+				finished: [student1],
+			});
+			const englishTask3 = taskFactory.build({
+				name: 'grammer2',
+				creator: teacher1,
+				lesson: grammerLesson,
+				users: [],
+			});
+			const englishTask4 = taskFactory.build({
+				name: 'grammer3',
+				creator: teacher1,
+				lesson: grammerLesson,
+			});
+
+			const historyTask1 = taskFactory.build({
+				name: 'cause of ww2',
+				course: historyCourse,
+				users: [student1, student2],
+			});
+
+			const historyTask2 = taskFactory.build({
+				name: 'fall of Rome',
+				course: historyCourse,
+				users: [student2, student3],
+				finished: [student2, student3],
+			});
+
+			const historyTask3 = taskFactory.build({
+				name: 'DDR culture',
+				course: historyCourse,
+			});
+
+			const mathsTask1 = taskFactory.build({
+				name: 'textbook page 12',
+				course: mathsCourse,
+				users: [student2],
+			});
+
+			const mathsTask2 = taskFactory.build({
+				name: 'all primes under million',
+				course: mathsCourse,
+				users: [student3],
+			});
+
+			const mathsTask3 = taskFactory.build({
+				name: 'algebra 1',
+				lesson: algebraLesson,
+				users: [student2],
+			});
+
+			return {
+				teacher1,
+				teacher2,
+				student1,
+				student2,
+				student3,
+				student4,
+				englishCourse,
+				grammerLesson,
+				historyCourse,
+				mathsCourse,
+				algebraLesson,
+				englishTask1,
+				englishTask2,
+				englishTask3,
+				englishTask4,
+				historyTask1,
+				historyTask2,
+				historyTask3,
+				mathsTask1,
+				mathsTask2,
+				mathsTask3,
+			};
+		};
+
+		beforeAll(async () => {
+			const module: TestingModule = await Test.createTestingModule({
+				imports: [ServerTestModule],
+			})
+				.overrideGuard(JwtAuthGuard)
+				.useValue({
+					canActivate(context: ExecutionContext) {
+						const req: Request = context.switchToHttp().getRequest();
+						req.user = currentUser;
+						return true;
+					},
+				})
+				.compile();
+
+			app = module.createNestApplication();
+			await app.init();
+			em = module.get(EntityManager);
+		});
+
+		afterAll(async () => {
+			await app.close();
+		});
+
+		beforeEach(async () => {
+			await cleanupCollections(em);
+			Configuration.set('FEATURE_TASK_CARD_ENABLED', false);
+		});
+
+		it('students 1 gets their tasks', async () => {
+			const entities = setup();
+
+			await em.persistAndFlush(Object.values(entities));
+			em.clear();
```

Comment:
looks like this 3 lines of code are repeated in all tests, so could be moved into beforeEach